GitUI/UserControls/RevisionGrid/Graph/RevisionGraphRevision.cs

```diff
@@ -94,7 +94,7 @@ public int EnsureScoreIsAbove(int minimalScore)
 
         public ObjectId Objectid { get; }
 
-        public ImmutableStack<RevisionGraphRevision> Parents => _parents;
+        public IEnumerable<RevisionGraphRevision> Parents => _startSegments.Select(segment => segment.Parent);
```

Comment:
_parents is ImmutableStack, _segments is ConcurrentQueue
But revisions are only updated from one thread, so it is probably OK.
ToArray() creates a snapshot, so it issafer.
Reading Parents in BranchFinder is not critical for performance, foreach in EnsureScoreIsAbove() is to some extent.
There is a lock issue still somewhere, why refresh is not running. I just want to avoid more issues...
But approving without requesting changes.

```suggestion
        public IEnumerable<RevisionGraphRevision> Parents => _startSegments.ToArray().Select(segment => segment.Parent);
```

Reply:
Who needs the safety of a snapshot, can call
`public RevisionGraphSegment[] GetStartSegments() => _startSegments.ToArray();`